Source/WebCore/dom/Element.cpp

```diff
@@ -2650,12 +2651,28 @@ Node::InsertedIntoAncestorResult Element::insertedIntoAncestor(InsertionType ins
             shadowRoot->hostChildElementDidChange(*this);
     }
 
-    if (!hasEffectiveLangState())
+    if (parentNode() == &parentOfInsertedTree && parentNode()->isDocumentNode()) {
```

Comment:
A more idiomatic way of checking whether parent is a document is `is<Document>(*parentNode())`.